chrome/views/popupView.js

```diff
@@ -63,68 +66,128 @@ GmailToTrello.PopupView.prototype.confirmPopup = function() {
+                    var due = new Date(card['due'].replace('T',' ') + ' UTC');
+                    if (minDue === null || due < minDue) {
+                        minDue = due;
+                    }
+                }
             }
+        });
+        if (allClosed) {
+            buttonText = 'Complete';
+        }
+        else if (!minDue) {
+            buttonText = threadCards.length + ' Card' + (threadCards.length === 1 ? '' : 's');
+        }
+        else if (minDue < new Date()) {
+            buttonText = 'Overdue';
+        }
+        else if (minDue.toString('yyyy-MM-dd') === Date.today().toString('yyyy-MM-dd')) {
+            buttonText = 'Today';
+        }
+        else if (minDue.toString('yyyy-MM-dd') === Date.parse('tomorrow').toString('yyyy-MM-dd')) {
+            buttonText = 'Tomorrow';
+        }
+        else {
+            buttonText = minDue.toString('MMM d, yyyy');
         }
-        gtt_log('PopupView:confirmPopup: adding Button and Popup');
-        $button.first().appendTo(this.$toolBar);
-        $popup.first().appendTo(this.$toolBar);
     }
 
-    if (needInit || $popup.length < 1) {
-        if (this.html && this.html['popup'] && this.html['popup'].length > 0) {
-            gtt_log('PopupView:confirmPopup: adding popup');
-            this.$toolBar.append(this.html['popup']);
-            needInit = true;
-        } else {
-            needInit = false;
-            $.get(chrome.extension.getURL('views/popupView.html'), function(data) {
-                // data = self.parent.replacer(data, {'jquery-ui-css': chrome.extension.getURL('lib/jquery-ui-1.12.1.min.css')}); // OBSOLETE (Ace@2017.06.09): Already loaded by manifest
-                self.html['popup'] = data;
-                gtt_log('PopupView:confirmPopup: creating popup')
-                self.$toolBar.append(data);
-                self.parent.loadSettings(self); // Calls init_popup
-            });
-        }
+    var img = 'GtT';
+    // Refresh icon present? If so, use graphics, if not, use text:
+    if ($('div.asl.T-I-J3.J-J5-Ji,div.asf.T-I-J3.J-J5-Ji', self.$toolBar).length > 0) {
+        img = '<img class="f tk3N6e-I-J3" src="'
+          + chrome.extension.getURL('images/trello-icon-black.png')
+          + '" />';
     }
 
-    if (needInit) {
-        this.parent.loadSettings(this); // Calls init_popup
+    self.html['add_to_trello'] =
+        '<div id="gttButton" class="T-I J-J5-Ji ar7 nf T-I-ax7 L3' + className + '" ' // "lS T-I-ax7 ar7"
+          + 'data-tooltip="Add this Gmail to Trello">'
+          + '<div aria-haspopup="true" role="button" class="J-J5-Ji W6eDmd L3 J-J5-Ji Bq L3" tabindex="0">' // class="J-J5-Ji W6eDmd L3 J-J5-Ji Bq L3">' // 
+          + img
+          + '<span id="gttButtonText"></span>'
+          + (buttonText.length > 0 ? '&nbsp;' + buttonText + '&nbsp;' : '')
+          + '<div id="gttDownArrow" class="G-asx T-I-J3 J-J5-Ji">&#9662;</div></div></div>';
+
+    self.confirmPopup();
+
+};
```

Comment:
This is the method to build the button HTML, which is a bit more complex now. It uses the new black/white trello icon and adds the class withCards if there are threadCards attached to this page. It also pulls some text to show on the button to simply indicate the status of the threadCards attached. It's probably easier to just read that logic than for me to explain it.